### PR TITLE
Add .NET Core to test project target frameworks

### PR DESCRIPTION
**Pull request by peterfreiling**
- Adding .NET Core 3.1 to test project target frameworks
- Refactoring test project .csproj format to the new cleaner format
- Fixing .NET Core specific bugs in codegen and multistring hashing
- Removing the TrillTest/ComponentTesting project which did nothing

### CODE REVIEW

File: Sources/Trill.sln [hunk 1]
```diff
@@ -43,20 +43,20 @@ Global
 		{3A2EB89B-70C7-4516-BB40-13248D5D2AB9}.Release|x64.Build.0 = Release|x64
 		{F88FBCF8-C33B-4089-8998-19E9CF901B99}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{F88FBCF8-C33B-4089-8998-19E9CF901B99}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{F88FBCF8-C33B-4089-8998-19E9CF901B99}.Debug|x64.ActiveCfg = Debug|x64
-		{F88FBCF8-C33B-4089-8998-19E9CF901B99}.Debug|x64.Build.0 = Debug|x64
+		{F88FBCF8-C33B-4089-8998-19E9CF901B99}.Debug|x64.ActiveCfg = Debug|Any CPU
```
Comment (arunkm) [inline Sources/Trill.sln]: Why some of `Debug|x64` map to `Debug|Any CPU` and some map to just `x64` ?

Comment (peterfreiling) [reply Sources/Trill.sln]: Hmm, I'm not sure, these changes were made automatically by VS. These two guids are the two remaining test projects, so I'm guessing .NET Core just targets Any CPU for both x64\AnyCPU

---
In reply to: [386604993](https://github.com/microsoft/Trill/pull/131#discussion_r386604993) [](ancestors = 386604993)

File: Sources/Test/Directory.Build.props [hunk 2]
```diff
@@ -1,110 +1,34 @@
...
   
-  <!-- This is needed for all .NET Framework projects that have a project reference to a multi-platform project: https://github.com/NuGet/Home/issues/4488 -->
-  <ItemGroup>
-    <PackageReference Include="Legacy2CPSWorkaround" Version="1.0.0" PrivateAssets="All" />
-  </ItemGroup>
-  
-  <!-- Setup global MSBuild variables -->
   <PropertyGroup>
-    <TargetFramework>net472</TargetFramework>
+    <Optimize>false</Optimize>
```
Comment (arunkm) [inline Sources/Test/Directory.Build.props]: Is this specific to `Debug` version ?

Comment (peterfreiling) [reply Sources/Test/Directory.Build.props]: This is just for test code, so I think it's fine to leave unoptimized for all flavors

---
In reply to: [386609180](https://github.com/microsoft/Trill/pull/131#discussion_r386609180) [](ancestors = 386609180)

File: Sources/Test/SimpleTesting/Common.cs [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+            var linqResult = enumerable.Where(wherePredicate.Compile()).Select(selectFunction.Compile());
+            var trillTemp = enumerable.ToStreamable();
+            if (disableFusion) trillTemp = trillTemp.AlterEventLifetime(s => s, StreamEvent.InfinitySyncTime);
+            var streamResult = trillTemp.Where(wherePredicate)
+                .Select(selectFunction)
+                .ToPayloadEnumerable();
+            return linqResult.SequenceEqual(streamResult);
+        }
+
+        public static IEnumerable<long> CountToInfinity(this long i) { var x = i; while (true) yield return x++; }
```
Comment (arunkm) [inline Sources/Test/SimpleTesting/Common.cs]: nit: add empty line after function, and probably some inside
(`Enumerable.Range(0, int.MaxValue)` also can be used if items are less than int.MaxValue)

Comment (peterfreiling) [reply Sources/Test/SimpleTesting/Common.cs]: Sure. FYI these are just copied/pasted.

---
In reply to: [386614803](https://github.com/microsoft/Trill/pull/131#discussion_r386614803) [](ancestors = 386614803)

**Pull request merged**